csp/templatetags/csp.py

```diff
@@ -19,10 +19,9 @@ def __init__(self, nodelist):
 
     def render(self, context):
         output = self.nodelist.render(context)
-        nonce = context.get('request')
+        request = context.get('request')
+        nonce = request.csp_nonce() if hasattr(request, 'csp_nonce') else ''
```

Comment:
I actually managed to shift this around a bit and lean on `SimpleLazyObject` to make the accessing `csp_nonce` way less ugly.